routes/pipeline.js

```diff
@@ -118,40 +121,27 @@ router.post('/pipeline_silence', common.ensureAuthenticated, function(req, res)
   var duration = data.duration;
   var until = null;
   var now = new Date();
-  var flavor = data.name.split('_')[0];
   if (duration == 'forever') {
-    mongo_db.get('pipelines').update({name: data.name}, {$set: {status: 'silent'}})
-    .then(() => res.json({}))
-    .catch(err => {console.log(err.message); return res.json({err: err.message});});
+    until = parseInt('-1');
   } else if (duration == 'monday') {
     var day = now.getDay();
-    if ((1 <= day) && (day <= 4)) {
-      // it's between Monday and Thursday
-      return res.json({err: 'Not available Monday-Thursday'});
-    }
+    if (day === 0) day = 7;  // make Sunday 7 instead of 0
     until = new Date();
```

Comment:
At least this only runs on one host (at least for us) but is it worth also being explicitly UTC here (I dunno how easy it is)?